pytheus/registry.py

```diff
@@ -0,0 +1,63 @@
+from threading import Lock
+from typing import Iterable, Protocol
+
+
+class Collector(Protocol):
+    name: str
+
+    def collect(self) -> Iterable:
+        pass
+
+
+class Registry(Protocol):
+    def register(self, collector: Collector):
+        pass
+
+    def unregister(self, collector: Collector):
+        pass
+
+    def collect(self) -> Iterable:
+        pass
+
+
+class CollectorRegistry:
+    def __init__(self, prefix: str = None) -> None:
+        self._lock = Lock()
+        self._prefix = prefix
+        self._collectors: dict[str, Collector] = {}
+
+    def register(self, collector: Collector) -> None:
+        with self._lock:
+            if collector.name in self._collectors:
+                return
+            self._collectors[collector.name] = collector
+
+    def unregister(self, collector: Collector) -> None:
+        with self._lock:
+            if collector.name not in self._collectors:
+                return
+            del self._collectors[collector.name]
+
+    def collect(self) -> Iterable:
+        for collector in self._collectors.values():
+            yield from collector.collect()
+
+
+class CollectorRegistryProxy:
```

Comment:
we had some dicussions on this, if we are able to remove this class completely and be able to set the registry as a configuration setting, it would be great!
But for a different PR

Reply:
+1 on this, we could use the same approach as with the Backend for consistency? (And have it configurable)

Happy for it to be done as a separate PR as well